cmd/pctl/main.go

```diff
@@ -41,22 +40,52 @@ func searchCmd() *cli.Command {
 	return &cli.Command{
 		Name:      "search",
 		Usage:     "search for a profile",
-		UsageText: "pctl --kubeconfig=<kubeconfig-path> search <QUERY>",
+		UsageText: "pctl [--kubeconfig=<kubeconfig-path>] search [--output table|json] <QUERY>",
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:        "output",
+				Aliases:     []string{"o"},
+				DefaultText: "table",
+				Value:       "table",
+				Usage:       "Output format. json|table",
+			},
+		},
 		Action: func(c *cli.Context) error {
 			searchName, catalogClient, err := parseArgs(c)
 			if err != nil {
 				_ = cli.ShowCommandHelp(c, "search")
 				return err
 			}
 
-			fmt.Printf("searching for profiles matching %q:\n", searchName)
 			profiles, err := catalog.Search(catalogClient, searchName)
 			if err != nil {
 				return err
 			}
-			for _, profile := range profiles {
-				fmt.Printf("%s: %s\n", profile.Name, profile.Description)
+
+			if len(profiles) == 0 {
+				fmt.Printf("No profiles found matching: '%s'\n", searchName)
+				return nil
+			}
+
+			var (
+				f      formatter.Formatter
+				getter func() interface{}
+			)
+
+			switch c.String("output") {
+			case "json":
+				f = formatter.NewJSONFormatter()
+				getter = func() interface{} { return profiles }
+			default:
+				f = formatter.NewTableFormatter()
+				getter = searchDataFunc(profiles)
 			}
```

Comment:
yeh i was thinking about this earlier. users could still do `-o table` tho, maybe if they don't know the default?

Reply:
But that's fine, because it just will be table, right? Anything else will just be anything else. :)

Reply:
wait is your snippet correct? should it be `if c.String("output") == "json"{`? 

because then that makes sense, `if c.String("output") {` is not possible (not bool) and is what made me question it

Reply:
yeah, sorry I just typed it in... it should be == "json"

Reply:
in that case yep, happy for the change :)